sub_peers.js

```diff
@@ -159,7 +240,9 @@ TChannelSubPeers.prototype.chooseHeapPeer = function chooseHeapPeer(req) {
     var self = this;
 
     var peer;
-    if (req && req.triedRemoteAddrs) {
+    if ((req && req.triedRemoteAddrs) ||
+        (this.hasMinConnections && this.currentConnectedPeers < this.minConnections)
```

Comment:
We only go into _choosePeerSkipTried() if we are below the threshold so the vast majority of the time we go in the hotpath of `choose()`

I'll optimize it if it turns out to be a real problem.